### PR TITLE
Smartnic doc

### PR DESCRIPTION
**Pull request by psairam**

### CODE REVIEW

File: Documentation/5.4.1/README.rst [hunk 1]
```diff
@@ -985,7 +1033,9 @@ where:
    * ``neutron-sriov.yaml`` Neutron SRIOV specific parameter values
    * ``avrs-role.yaml`` Enables services required for Compute Avrs role
    * ``ironic-role.yaml`` Enables Ironic Inspector service for Controller role
-
+   * ``ovs-hw-offload.yaml`` Enables OVS Hardware Offload on Smart NIC Compute nodes
```
Comment (shivsg) [inline Documentation/5.4.1/README.rst:1036]: Good to highlight this file as this is key for this feature.

File: Documentation/5.4.1/README.rst [hunk 2]
```diff
@@ -1568,6 +1645,43 @@ ironic-inspector.yaml for Ironic-Inspector Deployment
...
 
 
+ovs-hw-offload.yaml
+~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    # A Heat environment file that enables OVS Hardware Offload in the overcloud.
+    # This works by configuring SR-IOV NIC with switchdev and OVS Hardware Offload on
+    # compute nodes. The feature supported in OVS 2.8.0
```
Comment (shivsg) [inline Documentation/5.4.1/README.rst:1655]: Do we need to mention the OVS version here. Our VRS is OVS 2.5 based . I understand relevant code has been backported from upstream OVS 2.8 but still our base OVS remains the same.

File: tripleo-heat-templates-diff/common/diff_common_8.0.7-21 [hunk 3]
```diff
@@ -59,3 +59,80 @@ diff -r -u -N openstack-tripleo-heat-templates-original/docker/services/neutron-
...
+--- openstack-tripleo-heat-templates-original/puppet/services/neutron-compute-plugin-nuage.yaml 2019-01-31 00:27:45.770000000 -0500
++++ openstack-tripleo-heat-templates/puppet/services/neutron-compute-plugin-nuage.yaml  2019-01-31 00:26:21.594000000 -0500
+@@ -75,6 +75,25 @@
+     description: Support for non-default MTU configured on each Compute node
+     type: string
+     default: ''
++  OvsHwOffload:
++    default: false
++    description: |
++      Enable OVS Hardware Offload. This feature supported from OVS 2.8.0
```
Comment (shivsg) [inline tripleo-heat-templates-diff/common/diff_common_8.0.7-21:72]: Same as above. Good to leave out OVS version.

File: Documentation/5.4.1/README.rst [hunk 4]
```diff
@@ -1568,6 +1645,43 @@ ironic-inspector.yaml for Ironic-Inspector Deployment
...
+        #   2. You can pass your custom Tuned Profile to apply to the host
+        TunedProfileName: ""
+        OvsHwOffload: True
+        # Number of VFs that needs to be configured for a physical interface
+        NeutronSriovNumVFs: ["enp23s0f1:4:switchdev"]
+        # Mapping of SR-IOV PF interface to neutron physical_network.
+        # In case of Vxlan/GRE physical_network should be null.
+        # In case of flat/vlan the physical_network should as configured in neutron.
+        NovaPCIPassthrough:
+          - devname: "enp23s0f1"
```
Comment (shivsg) [inline Documentation/5.4.1/README.rst:1681]: How are multiple ports specified; example dual port NIC cards.

**Review by shivsg — APPROVED** [review]

File: Documentation/5.4.1/README.rst (hunk 2 above)
Comment (psairam) [reply Documentation/5.4.1/README.rst:1655]: @shivsg, this is an environment file that comes with openstack-tripleo-heat-templates package itself.
cc @sunny-verma

File: tripleo-heat-templates-diff/common/diff_common_8.0.7-21 (hunk 3 above)
Comment (psairam) [reply tripleo-heat-templates-diff/common/diff_common_8.0.7-21:72]: Ok, here we can make that change and I can remove "This feature supported from OVS 2.8.0" from the description
 cc @sunny-verma

File: Documentation/5.4.1/README.rst (hunk 4 above)
Comment (psairam) [reply Documentation/5.4.1/README.rst:1681]: @shivsg, currently we can only use 1 interface on a CX5 NIC for VRS offload. 

cc @sunny-verma @mathurvedant

Comment (shivsg) [reply Documentation/5.4.1/README.rst:1681]: Do we support 1 NIC or 1 port. The NIC we have has 2 ports.

**Pull request closed without merging**